admin/notifiers/class-subscription-notifier.php

```diff
@@ -0,0 +1,416 @@
+	 *
+	 * @var string
+	 */
+	const EXPIRED_MORE_THAN_1_DAY = 'plugins-expired-more-than-1-day';
+
+	/**
+	 * WPSEO_Subscription_Notifier constructor.
+	 */
+	public function __construct() {
+		$this->addon_manager       = $this->get_addon_manager();
+		$this->notification_center = $this->get_notification_center();
+	}
+
+	/**
+	 * Gets the addon manager.
+	 *
+	 * @return WPSEO_Addon_Manager
+	 *
+	 * @codeCoverageIgnore
+	 */
+	protected function get_addon_manager() {
+		return new WPSEO_Addon_Manager();
+	}
+
+	/**
+	 * Gets the notification center.
+	 *
+	 * @return Yoast_Notification_Center
+	 *
+	 * @codeCoverageIgnore
+	 */
+	protected function get_notification_center() {
+		return Yoast_Notification_Center::get();
+	}
+
+	/**
+	 * Initializes the subscription notification.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		$this->current_notification_id = get_option( self::NOTIFICATION_ID );
+
+		$this->subscription_with_earliest_expiry_date = $this->determine_subscription_with_earliest_expiry_date();
+
+		if ( $this->subscription_with_earliest_expiry_date === null ) {
+			$this->clean_up();
+			return;
+		}
+
+		$days_until_expiration = $this->calculate_days_until_expiration( $this->subscription_with_earliest_expiry_date );
+
+		$this->determine_notification( $days_until_expiration );
+	}
+
+	/**
+	 * Calculates the number of days until a subscription expires. 0 or below means the subscription has expired.
+	 *
+	 * @param stdClass $subscription Subscription to get the subscription expiration time for.
+	 *
+	 * @return int Number of days until expiration.
+	 */
+	protected function calculate_days_until_expiration( $subscription ) {
+		return (int) ceil( ( strtotime( $subscription->expiry_date ) - time() ) / DAY_IN_SECONDS );
+	}
+
+	/**
+	 * Sets the current notification to one of the defined constants, based on a set of conditions.
+	 *
+	 * @param int $days_until_expiration Days until the subscription expires.
+	 *
+	 * @return void
+	 */
+	private function determine_notification( $days_until_expiration ) {
+		if ( $days_until_expiration < 28 && $days_until_expiration > 7 ) {
+			$this->set_current_notification( self::EXPIRATION_WITHIN_4_WEEKS );
+
+			/* translators: %1$s expands to Yoast, %2$s expands to an opening anchor tag, %3$s expands to a percentage, %4$s expands to an closing anchor tag. */
+			$message = sprintf( __( 'Your %1$s plugin(s) will expire within 4 weeks. When plugins expire, you will no longer receive updates or support. %2$sRenew now to get a %3$s discount!%4$s', 'wordpress-seo' ), 'Yoast', '<a href="' . $this->get_url() . '"">', '25%', '</a>' );
+			$this->show_notification( $message );
+			return;
+		}
+		if ( $days_until_expiration <= 7 && $days_until_expiration > 1 ) {
+			$this->set_current_notification( self::EXPIRATION_WITHIN_1_WEEK );
+
+			$formatted_date = '<b>' . date_i18n( 'F jS, HH:MM', strtotime( $this->subscription_with_earliest_expiry_date->expiry_date ) ) . '</b>';
```

Comment:
Wrong time format:
<img width="296" alt="Screenshot 2019-04-01 at 11 29 10" src="https://user-images.githubusercontent.com/1682452/55317745-70ae0080-5471-11e9-9536-00b105b9ab74.png">

If it's meant to display hour and minutes, `HH:MM` should be changed to `H:i`. However, I'm not sure the date/time format should be hardcoded, as it needs to be different depending on the user locale. Maybe should use the one defined by users in the WP settings.

Also, the `S` in the date, forces the:
`S | English ordinal suffix for the day of the month, 2 characters | st, nd, rd or th`

to be always displayed also for non-English languages. See example below in Italian:

<img width="145" alt="Screenshot 2019-04-01 at 11 37 52" src="https://user-images.githubusercontent.com/1682452/55318335-d949ad00-5472-11e9-9fda-a6e9dca7a778.png">